viewer/db.js

```diff
@@ -1730,61 +1762,119 @@ Db.getIndices = async (startTime, stopTime, bounding, rotateIndex) => {
+          queryExtraIndexTimeMatched = true;
+          index = queryExtraIndexTimeMatch[0];
+        }
 
-      const start = Date.UTC(year, month - 1, day, hour) / 1000;
-      const stop = Date.UTC(year, month - 1, day, hour) / 1000 + len;
+        if (!queryExtraIndexTimeMatched){
+          // hourly 240311h19                     v year      v month        v day                    h  v hour
+          queryExtraIndexTimeMatch = iname.match(/([0-9][0-9])(0[1-9]|1[0-2])(0[1-9]|[12][0-9]|3[01])[Hh]([01][0-9]|2[0-3])$/);
+          if (queryExtraIndexTimeMatch) {
+            queryExtraIndexTimeMatched = true;
+            index = queryExtraIndexTimeMatch[0];
+          }
+        }
 
-      switch (bounding) {
-      default:
-      case 'last':
-        if (stop >= startTime && start <= stopTime) {
-          indices.push(iname);
+        if (!queryExtraIndexTimeMatched){
+            // weekly 24w10                       v year     w  v week
+          queryExtraIndexTimeMatch = iname.match(/([0-9][0-9])[Ww]([0-4][0-9]|5[0-3])$/);
+          if (queryExtraIndexTimeMatch) {
+            queryExtraIndexTimeMatched = true;
+            index = queryExtraIndexTimeMatch[0];
+          }
         }
-        break;
-      case 'first':
-      case 'both':
-      case 'either':
-      case 'database':
-        if (stop >= (startTime - len) && start <= (stopTime + len)) {
-          indices.push(iname);
+
+        if (!queryExtraIndexTimeMatched){
+            // monthly 24m10                      v year     w  v month
+          queryExtraIndexTimeMatch = iname.match(/([0-9][0-9])[Mm](0[1-9]|1[0-2])$/);
+          if (queryExtraIndexTimeMatch) {
+            queryExtraIndexTimeMatched = true;
+            index = queryExtraIndexTimeMatch[0];
+          }
         }
-        break;
+
+      } // if (isQueryExtraIndex)
+
+      if (!isQueryExtraIndex || queryExtraIndexTimeMatched) {
+
+        if (+index[0] >= 6) {
+          year = 1900 + (+index[0]) * 10 + (+index[1]);
+        } else {
+          year = 2000 + (+index[0]) * 10 + (+index[1]);
+        }
+
+        if (index[2] === 'w') {
+          len = 7 * 24 * 60 * 60;
+          month = 1;
+          day = (+index[3] * 10 + (+index[4])) * 7;
+        } else if (index[2] === 'm') {
+          month = (+index[3]) * 10 + (+index[4]);
+          day = 1;
+          len = 31 * 24 * 60 * 60;
+        } else if (index.length === 6) {
+          month = (+index[2]) * 10 + (+index[3]);
+          day = (+index[4]) * 10 + (+index[5]);
+          len = 24 * 60 * 60;
+        } else {
+          month = (+index[2]) * 10 + (+index[3]);
+          day = (+index[4]) * 10 + (+index[5]);
+          hour = (+index[7]) * 10 + (+index[8]);
+          // queryExtraIndices don't really have any way to specify (hourly[23468]|hourly12),
+          //   so for those hourly just means "hourly" with regards to length calculation
+          len = isQueryExtraIndex ? (60 * 60) : hlength;
```

Comment:
Arkime supports `hourly[23468]|hourly12` configurations for the `rotateIndex` value. In our regular expressions above trying to guess what type of index this is, we don't really have a good way to specify that. So for these `queryExtraIndices` index patterns we'll just have "hourly" mean "hourly" in these length calculations involved with determining search time frame vs. index time frame, without changing how it works for Arkime. Worst case scenario if we can't figure out what the time range is right below this in the `else if (isQueryExtraIndex) {` check where it will push the whole index pattern anyway.